include/cuda/std/span

```diff
@@ -11,12 +11,9 @@
 #ifndef _CUDA_SPAN
 #define _CUDA_SPAN
 
-#include "array"
```

Comment:
@jrhemstad those are the includes that we want to get rid of.

Ideally we would only include `__config` and the macro headers and then a single include of the internal header